### ISSUE TITLE
Re-ordering orders with custom options gives exception

### ISSUE THREAD

**Issue by jorgb90** [magento/magento2#22916]
Re-ordering orders with products with custom options isn't possible and gives "The product's required option(s) weren't entered. Make sure the options are entered and try again." as exception.

### Preconditions (*)
1. Magento 2.3.1

### Steps to reproduce (*)
1. Create a product with custom options
2. Create an order with this product with custom options
3. Try to re-order this order

### Expected result (*)
1. Order is recreated and can be placed

### Actual result (*)
1. After clicking on reorder we get the following exception: "The product's required option(s) weren't entered. Make sure the options are entered and try again."

**Comment by jorgb90** [comment]
Above problem is solved with this commit!  https://github.com/magento/magento2/pull/19272/files

**Comment by AlexWorking** [comment]
Hello @jorgb90. Unfortunately we've been not able to reproduce Your issue. Would You please try to reproduce it Yourself on the clear 2.3-develop version instance. You may ask for an instance as there is written above:
![Advised](https://user-images.githubusercontent.com/27814011/57920777-16b69e00-7859-11e9-8ecd-ddbca67b0fba.png)
If You manage to reproduce it, that will mean we've missed something crucial for reproduction and You will help us to find what it is. Thank You.

**Comment by jorgb90** [comment]
@AlexWorking Did you do the steps in the Magento 2 back-end? The initial order was created at the front-end and the re-order of that order was tried in the back-end. We have an almost clean 2.3.1 Magento install, but will try it with an instance here later on, but as said in my previous comment this commit https://github.com/magento/magento2/pull/19272/files fixed our issue.

**Comment by AlexWorking** [comment]
@jorgb90 Itried all the variations with the reorder including the one You've mentioned above - no result. Probably the issue was fixed with some commit (not the one You are talking about. That one wasn't merged). In order to close this issue or confirm it Your help is needed. Please try with @magento-engcom-team give me 2.3-develop instance - upcoming 2.3.1. Thank You.

**Comment by magento-engcom-team** [comment]
Hi @AlexWorking. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @AlexWorking, here is your [Magento instance](https://i-22916-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-22916-2-3-develop.instances.magento-community.engineering/admin](https://i-22916-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by sdzhepa** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @sdzhepa. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by sdzhepa** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @sdzhepa. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @sdzhepa, here is your [Magento instance](https://i-22916-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-22916-2-3-develop.instances.magento-community.engineering/admin](https://i-22916-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by sdzhepa** [comment]
Hello @jorgb90 

We tried to reproduce this issue on the vanilla Magento instance based on `2.3-develop` branch.

**Was used next steps**

1. Install clear vanilla Magento instance based on `2.3-develop`
2. Create test Simple product with several types custom options (required and not required)
-- [SCREEn of test product](https://www.screencast.com/t/nSFNZn3g9)
3. Go to StoreFront and add test product to the cart(selected only required options)
4. Go to Admin side _Sales > Orders_ and open newly create an order with test simple product
5. Click Reorder button
6.  Click Submit Order button

**Note**: Additionally was tested flow described in https://github.com/magento/magento2/issues/19267 and related PRs with the same result:
-- https://github.com/magento/magento2/issues/19267#issuecomment-489295457
-- https://github.com/magento/magento2/pull/19272#issuecomment-481676364

**Actual Result**
- Order successfully re-ordered
- no errors

We closed this issue with `Cannot Reproduce` resolution
but feel free to provide more details or missed steps/conditions and reopen it.
